### PR TITLE
Fix early stopping in converter patching + fix lr warmup for all tasks

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
Fix found bugs in OTX2.2 for OTX2.2.1 and Geti2.6 releases.

- [x] Correct EarlyStoppingWithWarmup overriding in converter instead of lightning early stopping. (I checked with Geti config that now it overrides early stopping correctly)
- [x] Add an interface for warming up scheduler in configs for all models. It should be possible to set warmup for every model. Next, warmup with 3 steps looks like a bug. This PR set it to 0 for now. (we need to search for better schedule later)
- [x] Change patience for ReduceLROnPlateue for classification task from 1 epoch to 5 (seems reasonable for me, 1 epoch looks very strange)
- [x] Increase patience for classification and align this default number across all current Geti tasks. (patience=3 are not reasonable and may hurt model convergence/performance due to random fluctuations.)
From Geti it is clearly seen, that sometimes more epochs are required to converge
image

- [x] Template update

image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — DISMISSED** [review]

**Comment by sovrasov** [comment]
@kprokofi do you have any experiments to compare the modified schedule for classification?

**Review by eugene123tw — DISMISSED** [review]

File: src/otx/recipe/rotated_detection/maskrcnn_efficientnetb2b.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ model:
     scheduler:
       class_path: otx.core.schedulers.LinearWarmupSchedulerCallable
       init_args:
-        num_warmup_steps: 100
```
Comment (eugene123tw) [inline src/otx/recipe/rotated_detection/maskrcnn_efficientnetb2b.yaml]: Hi @kprokofi could you revert the changes for rotated detection as well? Thanks

Comment (kprokofi) [reply src/otx/recipe/rotated_detection/maskrcnn_efficientnetb2b.yaml]: Done https://github.com/openvinotoolkit/training_extensions/pull/4131/commits/570014762261d4d02fa3e4667c4c1e587a28669c

**Comment by kprokofi** [comment]
After benchmarking all models, I found, that indeed updated schedule can show up to 2% of accuracy gain. However, the training time has grown drastically. For large datasets, I see 2-2.5x training time. So, I will decrease patience for classification from 10 to 5 epochs and patience for scheduler from 5 to 3.

Unfortunately, we don't have small datasets to test it on the edge cases like Geti, we have only one small dataset used for CI, but the accuracy becomes 1 after only few epochs.

**Comment by sovrasov** [comment]
@kprokofi could you update the changelog + also mention Dino V2 PR there?

**Comment by kprokofi** [comment]
> @kprokofi could you update the changelog + also mention Dino V2 PR there?

Done

**Review by sovrasov — DISMISSED** [review]

**Comment by kprokofi** [comment]
Classification tests:
image

**Review by eugene123tw — DISMISSED** [review]
Thanks, Kirill. This PR increases the early stopping patience at project level, which will affect all tasks. Let’s keep in mind that this might lead to increased accuracy but also training times in Geti, so we should monitor this closely.

**Comment by kprokofi** [comment]
> Thanks, Kirill. This PR increases the early stopping patience at project level, which will affect all tasks. Let’s keep in mind that this might lead to increased accuracy but also training times in Geti, so we should monitor this closely.

Eventually, It increases early stopping for classification only
patience=10 was always a default number in train.yaml
I also doesn't change it in templates for other tasks

image

**Review by sovrasov — APPROVED** [review]
